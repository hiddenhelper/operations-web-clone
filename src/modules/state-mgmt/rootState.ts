import { History } from 'history';

import { IState as IAuthState } from './auth/state';
import { IState as IGeneralState } from './general/state';
import { IState as IClientState } from './client/state';
import { IState as IProjectState } from './project/state';
import { IState as IUserState } from './user/state';
import { IState as IAccessControlSystemState } from './access-control-system/state';
import { IState as IBadgePrintingSystemState } from './badge-printer-system/state';
import { IState as IWorkerState } from './worker/state';
import { IState as ICertificationState } from './certification/state';
import { IState as IFileState } from './file/state';
import { IState as IBadgeState } from './badge/state';
import { IState as IStatisticsState } from './statistics/state';
import { IState as ITrainingState } from './training/state';
import { IState as IPaymentState } from './payment/state';
import { IState as IInvoiceState } from './invoice/state';
import { ApiService } from '../services/ApiService';
import { AuthService } from '../services/AuthService';
import { PrinterService } from '../services/PrinterService';
import { FileService } from '../services/FileService';
import { Logger } from '../services/Logger';
import { IProcoreState } from './procore/state';
import { IState as IProjectNewState } from './project-new/state';

export interface IAction {
  type: string;
  payload: any;
}

export interface IRootState {
  auth: IAuthState;
  user: IUserState;
  client: IClientState;
  project: IProjectState;
  accessControlSystem: IAccessControlSystemState;
  badgePrinterSystem: IBadgePrintingSystemState;
  worker: IWorkerState;
  certification: ICertificationState;
  training: ITrainingState;
  invoice: IInvoiceState;
  file: IFileState;
  badge: IBadgeState;
  statistics: IStatisticsState;
  payment: IPaymentState;
  general: IGeneralState;
  router?: { location: { pathname: string; search: string; hard: string; key: string }; action: string };
  procore: IProcoreState;
  projectNew: IProjectNewState;
}

export interface IEpicDependencies {
  history: History;
  apiService: ApiService;
  authService: AuthService;
  printerService: PrinterService;
  fileService: FileService;
  logger: typeof Logger;
}
