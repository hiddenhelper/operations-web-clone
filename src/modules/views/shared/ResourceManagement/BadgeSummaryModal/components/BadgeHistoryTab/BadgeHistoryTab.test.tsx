import React from 'react';
import { act, fireEvent, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createMockStore } from 'redux-test-utils';
import { MemoryRouter } from 'react-router';

import { getBadge_1, getBadgeHistory_1 } from '../../../../../../../test/entities';
import { getInitialState } from '../../../../../../../test/rootState';

import BadgeModal, { IBadgeHistoryTabProps } from './BadgeHistoryTab';

describe('BadgeHistoryTab', () => {
  global.console.warn = () => {
    /** */
  };
  let props: IBadgeHistoryTabProps;

  beforeEach(() => {
    props = {
      badgeId: getBadge_1().id,
      historyList: [getBadgeHistory_1()],
      count: 1,
      loading: undefined,
      fetchBadgeHistory: jest.fn(),
      clearBadgeHistory: jest.fn(),
    };
  });

  it('should loading', () => {
    props.loading = {
      isLoading: true,
      hasError: false,
      error: undefined,
    };
    const { container } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <BadgeModal {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it.skip('should render', () => {
    props.loading = { isLoading: false, error: null, hasError: false };
    const { container } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <BadgeModal {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });

  it('should change page', () => {
    props.loading = { isLoading: false, error: null, hasError: false };
    props.count = 17;
    const wrapper = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <BadgeModal {...props} />
        </MemoryRouter>
      </Provider>
    );

    const pageBtn = wrapper.getByText('2');

    act(() => {
      pageBtn.click();
    });

    expect(props.fetchBadgeHistory).toHaveBeenCalledWith(getBadge_1().id, { limit: 15, page: 2 });
  });

  it('should unMount', () => {
    const wrapper = render(
      <Provider store={createMockStore(getInitialState()) as any}>
        <MemoryRouter>
          <BadgeModal {...props} />
        </MemoryRouter>
      </Provider>
    );
    wrapper.unmount();
    expect(props.clearBadgeHistory).toHaveBeenCalled();
  });

  it('should render empty', () => {
    props.loading = { isLoading: false, error: null, hasError: false };
    props.historyList = [];
    props.count = 0;
    const { container } = render(
      <Provider store={createMockStore(getInitialState())}>
        <MemoryRouter>
          <BadgeModal {...props} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
