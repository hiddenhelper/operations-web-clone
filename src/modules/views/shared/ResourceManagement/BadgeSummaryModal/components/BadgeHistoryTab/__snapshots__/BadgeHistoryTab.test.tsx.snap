// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BadgeHistoryTab should loading 1`] = `
<div>
  <div
    style="padding: 15px;"
  >
    Loading...
  </div>
</div>
`;

exports[`BadgeHistoryTab should render empty 1`] = `
<div>
  <div
    style="padding: 15px;"
  >
    <div
      class="makeStyles-emptyBadgeHistory-153 makeStyles-container-171"
    >
      <div>
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-missingAvatarWrapper-182 makeStyles-missingAvatarBig-183 MuiAvatar-colorDefault"
        >
          <svg>
            e-badge.svg
          </svg>
        </div>
      </div>
      <p
        class="MuiTypography-root makeStyles-missingListText-172 MuiTypography-body1"
      >
        There is no Badge History
      </p>
    </div>
  </div>
</div>
`;
